### PR TITLE
feat(constraint): Add constraint type validation to bundle validation

### PR DESCRIPTION
**Pull request by dinhxuanvu**
**Description of the change:**

A new type of dependency named constraint (olm.constraint) which uses
CEL expression is introduced in OLM. This new type is added to
dependencies.json file in the bundle just like gvk and package
dependencies. As a result, a new validation for constraint type
is added to ensure this type will be validated for bundle
dependencies.

Signed-off-by: Vu Dinh <vudinh@outlook.com>

**Motivation for the change:**

**Reviewer Checklist**
- [ ] Implementation matches the proposed design, or proposal is updated to match implementation
- [ ] Sufficient unit test coverage 
- [ ] Sufficient end-to-end test coverage
- [ ] Docs updated or added to `/docs` 
- [ ] Commit messages sensible and descriptive

### CODE REVIEW

**Review by anik120 — CHANGES_REQUESTED** [review]
Looks good, a few test cases should get it ready to go.

File: pkg/registry/types.go [hunk 1]
```diff
@@ -289,6 +301,24 @@ func (pd *PackageDependency) Validate() []error {
...
 }
 
+// Validate will validate constraint type and return error(s)
+func (cc *CelConstraint) Validate() []error {
+	errs := []error{}
+	if cc.Cel == nil {
+		errs = append(errs, fmt.Errorf("The CEL field is missing"))
+	} else {
+		if cc.Cel.Rule == "" {
+			errs = append(errs, fmt.Errorf("The CEL expression is missing"))
```
Comment (estroz) [inline pkg/registry/types.go:311]:
```suggestion
			errs = append(errs, fmt.Errorf("The CEL expression is missing"))
			continue
```

File: pkg/registry/types.go [hunk 2]
```diff
@@ -329,6 +359,13 @@ func (e *Dependency) GetTypeValue() interface{} {
 			return nil
 		}
 		return dep
+	case ConstraintType:
+		dep := CelConstraint{}
```
Comment (estroz) [inline pkg/registry/types.go:363]: Shouldn't this be decoded into an api `constraints.Constraint`?

**Comment by estroz** [comment]
Agreed on unit tests.

**Pull request closed without merging**